lib/components/MonthPicker/MonthPicker.tsx

```diff
@@ -0,0 +1,252 @@
+    year: parseInt(year, 10),
+  };
+};
+
+type FieldType = keyof MonthPickerValue | 'native';
+const nativeToCustomValue = (
+  nativeValue: string,
+  value: MonthPickerValue,
+  fieldType: FieldType,
+) =>
+  ({
+    month: {
+      year: value && value.year ? value.year : undefined,
+      month: parseInt(nativeValue, 10) || undefined,
+    },
+    year: {
+      month: value && value.month ? value.month : undefined,
+      year: parseInt(nativeValue, 10) || undefined,
+    },
+    native: stringToCustomValue(nativeValue),
+  }[fieldType]);
+
+const handleChange = <Element extends HTMLSelectElement | HTMLInputElement>(
+  { onChange, value }: Pick<MonthPickerProps, 'onChange' | 'value'>,
+  fieldType: FieldType,
+) => (event: ChangeEvent<Element>) => {
+  if (typeof onChange === 'function') {
+    onChange(nativeToCustomValue(event.target.value, value, fieldType), event);
+  }
+};
+
+const handleBlur = <Element extends HTMLSelectElement | HTMLInputElement>(
+  { onBlur, value }: Pick<MonthPickerProps, 'onBlur' | 'value'>,
+  fieldType: FieldType,
+) => (event: FocusEvent<Element>) => {
+  if (typeof onBlur === 'function') {
+    const isRelatedTargetExternal =
+      event.relatedTarget !== monthRef.current &&
+      event.relatedTarget !== yearRef.current;
+
+    if (event.relatedTarget === null || isRelatedTargetExternal) {
+      onBlur(nativeToCustomValue(event.target.value, value, fieldType), event);
+    }
+  }
+};
+
+const handleFocus = <Element extends HTMLSelectElement | HTMLInputElement>({
+  onFocus,
+}: Pick<MonthPickerProps, 'onFocus'>) => (event: FocusEvent<Element>) => {
+  if (typeof onFocus === 'function') {
+    const isRelatedTargetExternal =
+      event.relatedTarget !== monthRef.current &&
+      event.relatedTarget !== yearRef.current;
+
+    if (event.relatedTarget === null || isRelatedTargetExternal) {
+      onFocus(event);
+    }
+  }
+};
+
+export const MonthPicker = ({
+  id,
+  name,
+  value,
+  label,
+  onChange,
+  onBlur,
+  onFocus,
+  tone,
+  disabled,
+  minYear = currYear - 100,
+  maxYear = currYear,
+  ascendingYears = false,
+  ...restProps
+}: MonthPickerProps) => {
+  const styles = useStyles(styleRefs);
+  const currentValue = {
+    month: value && value.month ? value.month : undefined,
+    year: value && value.year ? value.year : undefined,
+  };
+
+  const field = (
+    <Field
+      id={`${id}-month`}
+      name={`${name}-month`}
+      tone={tone}
+      disabled={disabled}
+      label={
+        <Box component={renderNativeInput ? 'span' : 'legend'}>
+          {label}
+          <Box component="span" display="none">
+            Month
+          </Box>
+        </Box>
+      }
+      {...restProps}
+      autoComplete={undefined}
+      secondaryMessage={null}
+    >
+      {({
+        backgroundColor,
+        boxShadow,
+        width,
+        paddingLeft,
+        paddingRight,
+        borderRadius,
+        className,
+        ...nonVisualFieldProps
```

Comment:
I wonder if this is a sign we've got the abstraction slightly wrong. Right now, `MonthPicker` needs to keep a running list of all visual props and keep them in sync with `Field`. In the future we might add another visual prop within `Field` and it would accidentally flow through into this `nonVisualFieldProps` object. Not sure what the answer is, though—might need to discuss this one.

Reply:
I noticed you added comments to `Field` to separate visual from non-visual props. I wonder if we should expose these as different objects, so consumers can safely ignore visual props if needed, e.g. `{ visualProps: { ... }, nonVisualProps: { ... } }`. Could obviously use a bikeshed.

Reply:
As discussed we are going to look at addressing this by introducing a `Fieldset` component, which suits this use case having multiple children components — whereas `Field` is really for one-to-one mappings